cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/config/ETLBatchConfig.java

```diff
@@ -30,33 +33,44 @@
   private final String schedule;
   private final List<ETLStage> actions;
 
+  // TODO consider using builder
```

Comment:
Why we have all these variance of constructors? Is it just for unit-test? At runtime of a pipeline, none of these would get called, right?